### PR TITLE
ASoC: SOF: amd: fix for false dsp interrupts

### PR DESCRIPTION
**Pull request by vijendarmukunda**
Fix for false dsp interrupts.

### CODE REVIEW

**Comment by vijendarmukunda** [comment]
@bhiregou @sunilkumardommati @plbossart  please help to review.

**Review by plbossart — APPROVED** [review]

**Review by iuliana-prodan — APPROVED** [review]

File: sound/soc/sof/amd/acp.c [hunk 1]
```diff
@@ -701,6 +701,10 @@ int amd_sof_acp_probe(struct snd_sof_dev *sdev)
 		goto unregister_dev;
 	}
 
+	ret = acp_init(sdev);
+	if (ret < 0)
+		goto free_smn_dev;
+
```
Comment (ranj063) [inline sound/soc/sof/amd/acp.c:707]: @vijendarmukunda just wondering if you need to call acp_reset() in the error paths below if you make this change

Comment (vijendarmukunda) [reply sound/soc/sof/amd/acp.c:707]: @ranj063 : not required. calling acp reset will reset all registers . we want to retain register values during failure.

**Review by ranj063 — APPROVED** [review]

**Review by sunilkumardommati — APPROVED** [review]

**Comment by plbossart** [comment]
SOFCI TEST

**Review by bhiregoudar — APPROVED** [review]

**Pull request merged**